src/azimuth/control/util.c

```diff
@@ -19,24 +19,68 @@
 
 #include "azimuth/control/util.h"
 
+#include <SDL_filesystem.h>
+
 #include <assert.h>
 #include <stdbool.h>
 #include <stdlib.h>
 
+#if !defined(__WINDOWS__) && !defined(__APPLE__)
+#include <string.h>
+#include <sys/stat.h>
+#include <wordexp.h>
+#endif
+
 #include "azimuth/gui/audio.h"
 #include "azimuth/state/save.h"
 #include "azimuth/system/resource.h"
 #include "azimuth/util/prefs.h"
 #include "azimuth/util/string.h"
 #include "azimuth/view/prefs.h"
 
+#define ORG_NAME NULL
+#define APP_NAME "Azimuth"
+
 /*===========================================================================*/
 
+// Get the path to the user-specific directory for storing persistent data for
+// this application (e.g. preferences or save files) as a NUL-terminated string.
+// If anything fails, this will return a NULL pointer.
+// Free the returned string with SDL_free().
+static char *az_get_app_data_directory(void) {
+#if !defined(__WINDOWS__) && !defined(__APPLE__)
+  // First, do tilde-expansion so we get a path in the user's homedir.
+  wordexp_t words;
+  wordexp("~/.azimuth-game", &words, 0);
+  if (words.we_wordc < 1) {
+    wordfree(&words);
+    return NULL;
+  }
+  const size_t exp_len = strlen(words.we_wordv[0]);
+  char* path_buffer = (char*)SDL_calloc(1, exp_len + 1);
+  if (NULL == path_buffer)
+  {
```

Comment:
Minor nitpick: for consistency with the rest of the codebase, the opening brace should go on the same line as the `if` statement.